.github/workflows/linter.yml

```diff
@@ -1,31 +1,71 @@
----
-name: Lint Code Base
+name: Lint
 
 on:
   pull_request:
-    branches: [ master ]
+    branches:
+      - master
+
+permissions:
+  checks: write
+  contents: write
 
 jobs:
-  build:
-    name: Lint Code Base
+  lint-python:
+    name: Lint Python
     runs-on: ubuntu-latest
 
-    permissions:
-      contents: read
-      packages: read
-      statuses: write
+    steps:
+      - name: Check out Git repository
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Check changed py files
+        id: changed-files
+        uses: tj-actions/changed-files@v24
+        with:
+          files: |
+            **/*.py
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        if: ${{ steps.changed-files.outputs.all_changed_files }}
+        with:
+          python-version: 3.8
```

Comment:
commcare-cloud requires Python 3.10. Would it make more sense to use that version?